rules/dead-code/tests/Rector/Class_/RemoveUnusedClassesRector/Fixture/skip_class_extend.php.inc

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Rector\DeadCode\Tests\Rector\Class_\RemoveUnusedClassesRector\Fixture;
+
+class Car extends Vehicle
+{
+    /**
+     * @var string
+     */
+    private $engineSize;
+
+    public function run(): void
+    {
+        echo (new self())->hasEngine();
+    }
+}
+
+class Vehicle
+{
+    /**
+     * @var string
+     */
+    protected $colour;
+
+    /**
+     * @var bool
+     */
+    protected $engine;
+
+    public function hasEngine(): bool
+    {
+        return $this->engine;
+    }
+}
```

Comment:
This file should not be in `/Fixture/`, but in `/Source/` directory, as it's not the tested class

By convention, the test fixture should only contain tested class

Reply:
If I move the Vehicle class to source, then the tests are green.

Reply:
Ah, so it's not about class that extends, but class that has children, right?
I got confused by the fixture name.

In that case the `Vehicle` should be moved first, and named `SkipClassThatHasUsedChildren`. Then reflection check on:

```
$classReflection = $scope->getClassReflection();

if ($classReflection->getAncestors()) {
    return [];
}
```

Reply:
Then reflection check on:

```php
$classReflection = $scope->getClassReflection();

if ($classReflection->getAncestors()) {
    return [];
}
```

Not sure what you mean here? Where should this be done/checked?

Reply:
Probably here https://github.com/rectorphp/rector/blob/9f6834c46ea7599446dce808d21a069a4a6a9189/rules/dead-code/src/Rector/Class_/RemoveUnusedClassesRector.php#L118